diffsims/simulations/simulation2d.py

```diff
@@ -0,0 +1,746 @@
+                self.phase_index += 1
+            else:
+                self.rotation_index += 1
+            return coords
+
+    @property
+    def current_size(self):
+        """Returns the number of rotations in the current phase"""
+        if self.has_multiple_phases:
+            return self.rotations[self.phase_index].size
+        else:
+            return self.rotations.size
+
+    def deepcopy(self):
+        return copy.deepcopy(self)
+
+    def _get_transformed_coordinates(
+        self,
+        angle: float,
+        center: Sequence = (0, 0),
+        mirrored: bool = False,
+        units: str = "real",
+        calibration: float = None,
+    ):
+        """Translate, rotate or mirror the pattern spot coordinates"""
+
+        coords = self.get_current_coordinates()
+
+        if units != "real":
+            center = np.array(center)
+            coords.data = coords.data / calibration
+        transformed_coords = coords
+        cx, cy = center
+        x = transformed_coords.data[:, 0]
+        y = transformed_coords.data[:, 1]
+        mirrored_factor = -1 if mirrored else 1
+        theta = mirrored_factor * np.arctan2(y, x) + np.deg2rad(angle)
+        rd = np.sqrt(x**2 + y**2)
+        transformed_coords[:, 0] = rd * np.cos(theta) + cx
+        transformed_coords[:, 1] = rd * np.sin(theta) + cy
+        return transformed_coords
+
+    @property
+    def current_phase(self):
+        if self.has_multiple_phases:
+            return self.phases[self.phase_index]
+        else:
+            return self.phases
+
+    def rotate_shift_coordinates(
+        self, angle: float, center: Sequence = (0, 0), mirrored: bool = False
+    ):
+        """Rotate, flip or shift patterns in-plane
+
+        Parameters
+        ----------
+        angle
+            In plane rotation angle in degrees
+        center
+            Center coordinate of the patterns
+        mirrored
+            Mirror across the x-axis
+        """
+        coords_new = self._get_transformed_coordinates(
+            angle, center, mirrored, units="real"
+        )
+        return coords_new
+
+    def polar_flatten_simulations(self, radial_axes=None, azimuthal_axes=None):
+        """Flattens the simulations into polar coordinates for use in template matching.
+        The resulting arrays are of shape (n_simulations, n_spots) where n_spots is the
+        maximum number of spots in any simulation.
+
+
+        Returns
+        -------
+        r_templates, theta_templates, intensities_templates
+        """
+
+        flattened_vectors = [sim for sim in self]
+        max_num_spots = max([v.size for v in flattened_vectors])
+
+        r_templates = np.zeros((len(flattened_vectors), max_num_spots))
+        theta_templates = np.zeros((len(flattened_vectors), max_num_spots))
+        intensities_templates = np.zeros((len(flattened_vectors), max_num_spots))
+        for i, v in enumerate(flattened_vectors):
+            r, t, _ = v.to_polar()
```

Comment:
I am getting some strange results when I try to template match, and I believe this line is the cause. Should it not be `t, _, r = v.to_polar()`? Seeing as `Vector3d.to_polar` returns (azimuthal, polar, radius) spherical coordinates. By inspection, the current way yields `t`-values close to pi/2, which would make sense for the polar angle with a mostly flat Ewald's sphere.

Reply:
@viljarjf Sorry, I've gotten kind of confused with all of these comments and changes.  We don't want to use that function at all... We want the radius of the projection of the vector onto the x,y plane.

Reply:
@viljarjf This (should) be better